src/prover.rs

```diff
@@ -144,24 +144,27 @@ where
     })
 }
 
-/// Given a Groth16 proof, returns a proof of the same statement. The proof that is returned is
-/// computationally unlinkable to the given proof, assuming the RNG is cryptographically secure
+/// Given a Groth16 proof, returns a fresh proof of the same statement. For a proof π of a
+/// statement S, the output of the non-deterministic procedure `rerandomize_proof(π)` is
+/// statistically indistinguishable from a fresh honest proof of S. For more info, see theorem 3 of
+/// BKSV20: https://eprint.iacr.org/2020/811
 pub fn rerandomize_proof<E, R>(rng: &mut R, vk: &VerifyingKey<E>, proof: &Proof<E>) -> Proof<E>
 where
     E: PairingEngine,
     R: Rng,
 {
     // These are our rerandomization factors
-    let (r, s) = (E::Fr::rand(rng), E::Fr::rand(rng));
+    let (r1, r2) = (E::Fr::rand(rng), E::Fr::rand(rng));
```

Comment:
Let's sample in a way that ensures that `r1` and `r2` can never be zero.

```suggestion
    let (mut r1, mut r2) = (E::Fr::zero(), E::Fr::zero());
    while r1.is_zero() || r2.is_zero() {
    	r1 = E::Fr::rand(rng);
    	r2 = E::Fr::rand(rng);
    }
```

Reply:
lgtm!